web/src/containers/admin/CreateUser.js

```diff
@@ -0,0 +1,223 @@
+import { Link } from 'react-router-dom';
+import zxcvbn from 'zxcvbn';
+import { STATES } from '../../util';
+
+import { createUser as createUserDispatch } from '../../actions/admin';
+
+import Btn from '../../components/Btn';
+import { t } from '../../i18n';
+
+class CreateUser extends Component {
+  state = {
+    fetching: false,
+    name: '',
+    email: '',
+    password: '',
+    passwordScore: 0,
+    passwordQuality: 'Weak',
+    showPassword: false,
+    state: ''
+  };
+
+  toggleShowPassword = () => {
+    this.setState({ showPassword: !this.state.showPassword });
+  };
+
+  handleChange = e => {
+    const { name, value } = e.target;
+    const change = { [name]: value };
+
+    if (name === 'password') {
+      const score = zxcvbn(value, [this.state.email, this.state.name]);
+
+      // guesses_log10 is the log10 of the number of guesses zxcvbn estimates
+      // would be necessary to find the password.  10^8 is considered a score
+      // of 3, which is what we're using.  But guesses_log10 is a decimal
+      // number, which means it can have weird junk in it due to rounding
+      // issues.  The math here is basically rounding to 4 decimal places and
+      // then dividing by 8.5 to get a percentage of how "acceptable" the
+      // password is.  Since a log10 of 8.5 would meet our criteria, that
+      // corresponds with 100%.  I then cap it with Math.min to make sure we
+      // don't end up with more than 100%.
+      change.passwordScore = Math.min(
+        1,
+        Math.floor(score.guesses_log10 * 10000) / 85000
+      );
+
+      change.passwordQuality = score.score > 2 ? 'Good' : 'Weak';
+    }
+
+    this.setState(change);
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    const { email, name, password, state } = this.state;
+    if (!email || !password || !state) {
+      alert('Email, password, and state are all required');
+      return;
+    }
+
+    const score = zxcvbn(password, [email, name]);
+    if (score.score < 3) {
+      alert('Password is too weak');
+      return;
+    }
+
+    const { createUser } = this.props;
+
+    this.setState({ fetching: true });
+    createUser({ email, name, password, state })
+      .then(() => {
+        this.setState({
+          fetching: false,
+          name: '',
+          email: '',
+          password: '',
+          passwordScore: 0,
+          passwordQuality: 'Weak',
+          state: ''
+        });
+      })
+      .catch(() => {
+        this.setState({ fetching: false });
+      });
+  };
+
+  render() {
+    const {
+      fetching,
+      name,
+      email,
+      password,
+      passwordScore,
+      passwordQuality,
+      showPassword,
+      state
+    } = this.state;
+
+    // The password percent bar thing is 88% width, so scale
+    // the password score thing accordingly.  That's all.
+    const passwordPercent = 88 * passwordScore;
+
+    // Really the whole password meter should be replaced.
```

Comment:
We got an issue for that?

Reply:
We do *not*, yes!  😬  Will add that issue.